src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/VaultContentXMLContentCreator.java

```diff
@@ -0,0 +1,366 @@
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createGroup(String name, String[] members, Map<String, Object> extraProperties) throws RepositoryException {
+        try {
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB  .append("create group " + name + " \n\n")
+                            .append("add " + String.join(",", members) + " to group " + name)
+                            .append("set properties on authorizable(" + name + ")\n");
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createAce(String principal, String[] grantedPrivileges, String[] deniedPrivileges, String order) throws RepositoryException {
+        try {
+            //set principal ACL for principal1,principal2
+            //allow jcr:read
+            StringBuilder repoInitTextSB = new StringBuilder();
+            String path = this.currentNode.getPath();
+            
+            repoInitTextSB.append("set ACL for " + principal +"\n\n");
+         
+            for(String privilege: grantedPrivileges){
+                repoInitTextSB.append("allow " + privilege + " on " + path + "\n");
+            }
+            for(String privilege: deniedPrivileges){
+                repoInitTextSB.append("deny " + privilege + " on " + path + "\n");
+            }
+            repoInitTextSB.append("end");
+            
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createAce(String principalId, String[] grantedPrivilegeNames, String[] deniedPrivilegeNames,
+                          String order, Map<String, Value> restrictions, Map<String, Value[]> mvRestrictions,
+                          Set<String> removedRestrictionNames) throws RepositoryException {
+        try {
+            //set principal ACL for principal1,principal2
+            //allow jcr:read
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB.append("set principal ACL for " + principalId +"\n\n");
+            String path = this.currentNode.getPath();
+            if(grantedPrivilegeNames != null){
```

Comment:
- please reformat the code that whitespace is consistent (here and below is an extra space missing)
- this whole block should IMHO be removed.... it doesn't make sense if you pass restrictions